src/python/dxpy/scripts/dx_build_app.py

```diff
@@ -1022,14 +1022,26 @@ def get_project_to_check():
         if "project" in extra_args:
             return extra_args["project"]
         if destination:
```

Comment:
here I think since destination already has a default value, we can skip the if condition

also, I would suggest not changing this function, but instead checking it right before we launch the api call.

Reply:
> here I think since destination already has a default value, we can skip the if condition
> 
> also, I would suggest not changing this function, but instead checking it right before we launch the api call.

You mean, right before the applet-xxxx/build?

Reply:
Oh sorry I need to correct it a little here:

I was wondering if we happen to use destination something like `:` which is resolved to a container id, then the following if condition will not be satisfied. 
`if destination == "." or destination.startswith("container-"):`

so I would suggest, maybe we still parse_destination and let this function return the get_project_to_check, then before the checkFeatureAccess request, we check if we are in a execution env with a DX_JOB_ID already set and this "project_to_check" is a valid one